application/src/main/java/org/thingsboard/server/service/queue/DefaultTbClusterService.java

```diff
@@ -394,6 +400,20 @@ private void broadcast(ToTransportMsg transportMsg, TbQueueCallback callback) {
         }
     }
 
+    private void broadcast(ToTransportMsg transportMsg, String transportType, TbQueueCallback callback) {
+        TbQueueProducer<TbProtoQueueMsg<ToTransportMsg>> toTransportNfProducer = producerProvider.getTransportNotificationsMsgProducer();
+        Set<TransportProtos.ServiceInfo> tbTransportInfos = partitionService.getAllServices(ServiceType.TB_TRANSPORT);
+        TbQueueCallback proxyCallback = callback != null ? new MultipleTbQueueCallbackWrapper(tbTransportInfos.size(), callback) : null;
```

Comment:
tbTransportInfos.size() might be greater than filtered size. filter first and than create a callback with a right size